src/main/java/com/smartsheet/wiremock/extensions/ApiScenarioTransformer.java

```diff
@@ -52,22 +45,27 @@ public ResponseDefinition transform(
 			FileSource files,
 			Parameters parameters) {
 
-		if (isMatched(responseDefinition)) return responseDefinition;
 		if (!scenarioHeaderIsValid(request)) return INVALID_SCENARIO_RESPONSE;
 
 		JsonNode scenario = getScenario(request, files);
 		if (scenario == null) return buildUnknownScenarioResponse(request);
 
-		return getDiffResponse(request, scenario);
+		String diff = RequestDiff.getDiff(request, scenario);
+
+		if (isMatched(responseDefinition) && hasNoDiff(diff)) return responseDefinition;
```

Comment:
This is the main change. Rather than letting all matched responses through, we are now also verifying that our diff returned empty. This allows us to add the unexpected query param check